### PR TITLE
Adding Tournament Team APIs

### PR DESCRIPTION
**Pull request by adityatMS**

### CODE REVIEW

File: Source/Services/Tournaments/tournament_service.cpp [hunk 1]
```diff
@@ -145,6 +146,118 @@ tournament_service::get_tournament_details(
...
 }
 
+pplx::task<xbox::services::xbox_live_result<team_request_result>>
+tournament_service::_Get_teams(
+    _In_ const string_t& nextLinkUrl
+    )
+{
+    auto subPath = L"";
+    return get_teams_internal(subPath);
+}
```
Comment (jasonsandlin) [inline Source/Services/Tournaments/tournament_service.cpp]: is this right?

File: Source/Services/Tournaments/WinRT/CurrentMatchMetadata_WinRT.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    /// The uniquely identifying information for the game session.
+    /// </summary>
+    property Microsoft::Xbox::Services::Multiplayer::MultiplayerSessionReference^ GameSessionReference 
+    { 
+        Microsoft::Xbox::Services::Multiplayer::MultiplayerSessionReference^ get(); 
+    }
+
+    /// <summary>
+    /// 
+    /// </summary>
```
Comment (jasonsandlin) [inline Source/Services/Tournaments/WinRT/CurrentMatchMetadata_WinRT.h]: todo

File: Source/Services/Tournaments/WinRT/PreviousMatchMetadata_WinRT.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    /// <summary>
+    /// The timestamp when the match ended.
+    /// </summary>
+    DEFINE_PROP_GET_DATETIME_OBJ(MatchEndTime, match_end_time);
+
+    /// <summary>
+    /// 
+    /// </summary>
```
Comment (jasonsandlin) [inline Source/Services/Tournaments/WinRT/PreviousMatchMetadata_WinRT.h]: todo

File: Source/Services/Tournaments/WinRT/TeamCompletedReason_WinRT.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
+//
+//*********************************************************
+#pragma once 
+
+NAMESPACE_MICROSOFT_XBOX_SERVICES_TOURNAMENTS_BEGIN
+
+/// <summary>
+/// 
+/// </summary>
```
Comment (jasonsandlin) [inline Source/Services/Tournaments/WinRT/TeamCompletedReason_WinRT.h]: todo here and below

File: Source/Services/Tournaments/WinRT/TeamOrderBy_WinRT.h [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
+//
+//*********************************************************
+#pragma once 
+
+NAMESPACE_MICROSOFT_XBOX_SERVICES_TOURNAMENTS_BEGIN
+
+/// <summary>
+/// 
+/// </summary>
```
Comment (jasonsandlin) [inline Source/Services/Tournaments/WinRT/TeamOrderBy_WinRT.h]: todo

File: Source/Services/Tournaments/WinRT/TeamCompletedReason_WinRT.h [hunk 6]
```diff
@@ -0,0 +1,41 @@
+//*********************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
+// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
+// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
+// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
+//
+//*********************************************************
```
Comment (adityatMS) [inline Source/Services/Tournaments/WinRT/TeamCompletedReason_WinRT.h]: Change headers to match the new copyright info.

File: Tests/UnitTests/Tests/Services/TournamentsTests.cpp [hunk 7]
```diff
@@ -1,6 +1,12 @@
...
+//*********************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
+// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
+// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
+// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
+//
+//*********************************************************
 #include "pch.h"
```
Comment (jasonsandlin) [inline Tests/UnitTests/Tests/Services/TournamentsTests.cpp]: revert header

File: Source/Services/Tournaments/tournament_service.cpp (hunk 1 above)
Comment (adityatMS) [inline Source/Services/Tournaments/tournament_service.cpp]: Will fix this in my next check-in as it requires some wiring that I haven't figured out yet.

---
In reply to: [106765896](https://github.com/Microsoft/xbox-live-api/pull/103#discussion_r106765896) [](ancestors = 106765896)

**Pull request merged**